### PR TITLE
identifier function and related code to deprecated

### PR DESCRIPTION
**Pull request by bkamins**
This PR cleans up some deprecated functionality and moves it to deprecated section.

@nalimilan also `colmissing` and `countmissing` functions are not used in DataFrames.jl except one place in tests and they are not exported. Do we want to keep them?

### CODE REVIEW

**Comment by bkamins** [comment]
The drop in CI is purely due to moving code around.

**Comment by nalimilan** [comment]
> @nalimilan also `colmissing` and `countmissing` functions are not used in DataFrames.jl except one place in tests and they are not exported. Do we want to keep them?

Not, let's remove them if they aren't used.

**Comment by bkamins** [comment]
CI failures seem unrelated.

File: src/deprecated.jl [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import Base: isidentifier, is_id_start_char, is_id_char
```
Comment (nalimilan) [inline src/deprecated.jl]: I hadn't noticed this. Better fully qualify these, so that we can remove the deprecations separately without having unrelated code at the top of the file.

Comment (bkamins) [reply src/deprecated.jl]: fixed
